js_modules/dagit/packages/core/src/app/WebSocketProvider.tsx

```diff
@@ -96,6 +102,23 @@ export const WebSocketProvider: React.FC<Props> = (props) => {
     };
   }, [debouncedSetter, websocketClient]);
 
+  // Wait a little while for the server to ack the WebSocket connection. If it never
+  // acks, never closes, and never errors, we shouldn't keep the app waiting to connect
+  // forever. Instead, set WebSocket availability as `unavailable` and let use cases
+  // fall back to non-WS implementations.
+  React.useEffect(() => {
+    let timeout: ReturnType<typeof setTimeout> | null = null;
+    if (availability === 'attempting-to-connect') {
+      timeout = setTimeout(() => {
+        setAvailability('unavailable');
```

Comment:
Should we log something here in the console so that we can debug when this happens unexpectedly?